app/src/main/java/com/itsaky/androidide/fragments/ProjectListFragment.kt

```diff
@@ -0,0 +1,143 @@
+/*
+ *  This file is part of AndroidIDE.
+ *
+ *  AndroidIDE is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  AndroidIDE is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide.fragments
+
+import android.os.Bundle
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.method.LinkMovementMethod
+import android.text.style.ClickableSpan
+import android.view.View
+import androidx.annotation.StringRes
+import androidx.fragment.app.viewModels
+import com.google.android.flexbox.FlexDirection
+import com.google.android.flexbox.FlexboxLayoutManager
+import com.google.android.flexbox.JustifyContent
+import com.itsaky.androidide.R
+import com.itsaky.androidide.activities.MainActivity
+import com.itsaky.androidide.adapters.ProjectListAdapter
+import com.itsaky.androidide.databinding.FragmentProjectListBinding
+import com.itsaky.androidide.models.ProjectInfoDetails
+import com.itsaky.androidide.models.toProjectInfoDetails
+import com.itsaky.androidide.provider.IDEViewModelProvider
+import com.itsaky.androidide.utils.ProjectInfoDetailsUtils
+import com.itsaky.androidide.viewmodel.MainViewModel
+import com.itsaky.androidide.viewmodel.ProjectInfoViewModel
+import java.io.File
+
+
+class ProjectListFragment :
+  FragmentWithBinding<FragmentProjectListBinding>(R.layout.fragment_project_list,
+    FragmentProjectListBinding::bind) {
+
+  private var adapter: ProjectListAdapter? = null
+  private var layoutManager: FlexboxLayoutManager? = null
+
+  private val viewModel by viewModels<MainViewModel>(ownerProducer = { requireActivity() })
+  private val projectInfoViewModel: ProjectInfoViewModel by viewModels { IDEViewModelProvider.Factory }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    setupRecyclerView()
+    observeProjectInfoList()
+
+    projectInfoViewModel.getAllProjectInfo()
+
+    binding.exitButton.setOnClickListener {
+      viewModel.setScreen(MainViewModel.SCREEN_MAIN)
+    }
+
+  }
+
+  private fun setupRecyclerView() {
+    layoutManager = FlexboxLayoutManager(requireContext(), FlexDirection.ROW)
```

Comment:
Why you need FlexboxLayoutManagerr. Use a LinearLayoutManager with vertical orientation

Reply:
I was going to do the same thing with the model lists, but I decided against it.